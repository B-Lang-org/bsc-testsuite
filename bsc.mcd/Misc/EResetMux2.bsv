import Clocks::*;

(* synthesize *)
module sysEResetMux2(Clock c, Reset r1, Reset r2, SyncBitIfc#(Bit#(1)) out_ifc);
  Reg#(Bool) b <- mkReg(False);
 
  // illegally muxed reset
  Reset r = b ? r1 : r2;
  
  SyncBitIfc#(Bit#(1)) ifc();
  mkSyncBitToCC#(c, r) the_ifc(ifc);
  
  return ifc; 
endmodule
