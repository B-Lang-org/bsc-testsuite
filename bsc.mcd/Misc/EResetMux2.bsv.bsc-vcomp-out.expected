checking package dependencies
compiling EResetMux2.bsv
code generation for sysEResetMux2 starts
Error: "EResetMux2.bsv", line 11, column 3: (G0072)
  Input reset `sRstIn' of submodule `the_ifc' is instantiated with a dynamic
  expression. Input resets must be determined at compile time.
  During elaboration of `the_ifc' at "EResetMux2.bsv", line 11, column 3.
  During elaboration of `sysEResetMux2' at "EResetMux2.bsv", line 4, column 8.
